src/software/ai/hl/stp/tactic/dribble/dribble_fsm.h

```diff
@@ -0,0 +1,220 @@
+                event.common.robot.id(), intercept_position, face_ball_orientation, 0,
+                DribblerMode::MAX_FORCE, BallCollisionType::ALLOW,
+                AutoChipOrKick{AutoChipOrKickMode::OFF, 0},
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT));
+        };
+
+        /**
+         * Action to update to get possession of the ball
+         *
+         * @param event DribbleBallFSM::Update
+         */
+        const auto dribble = [this, ball_at_dest](auto event) {
+            // Threshold to determine if the robot has the expected orientation
+            static const Angle ROBOT_ORIENTATION_CLOSE_THRESHOLD = Angle::fromDegrees(5);
+            // helper calculations
+            auto ball_position    = event.common.world.ball().position();
+            auto ball_destination = getBallDestination(event);
+            Angle to_destination_orientation =
+                (ball_destination - ball_position).orientation();
+
+            // Default destination and orientation assume ball is at the destination
+            // pivot to final face ball destination
+            Angle target_orientation = getFinalFaceBallOrientation(event);
+            Point target_destination =
+                convertBallPositionToRobotPosition(ball_destination, target_orientation);
+
+            if (!ball_at_dest(event))
+            {
+                if (to_destination_orientation.minDiff(event.common.robot.orientation()) <
+                    ROBOT_ORIENTATION_CLOSE_THRESHOLD)
+                {
+                    // dribble towards ball destination
+                    target_destination = convertBallPositionToRobotPosition(
+                        ball_destination, to_destination_orientation);
+                    target_orientation = to_destination_orientation;
+                }
+                else
+                {
+                    // pivot to face ball destination
+                    target_destination = convertBallPositionToRobotPosition(
+                        ball_position, to_destination_orientation);
+                    target_orientation = to_destination_orientation;
+                }
+            }
+            event.common.set_intent(std::make_unique<MoveIntent>(
+                event.common.robot.id(), target_destination, target_orientation, 0,
+                DribblerMode::MAX_FORCE, BallCollisionType::ALLOW,
+                AutoChipOrKick{AutoChipOrKickMode::OFF, 0},
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT));
+        };
+
+        return make_transition_table(
+            // src_state + event [guard] / action = dest_state
+            *get_possession_s + update_e[have_possession] / dribble = dribble_s,
+            get_possession_s + update_e[!have_possession] / get_possession,
+            dribble_s + update_e[!have_possession] / get_possession = get_possession_s,
+            dribble_s + update_e[!ball_at_dest] / dribble,
+            dribble_s + update_e[ball_at_dest] / dribble    = X,
```

Comment:
We need something here to prevent the user of this FSM from requesting a dribble distance that can break the rules.

Maybe a slight kick every meter?